Rubberduck.Parsing/VBA/ModuleState.cs

```diff
@@ -20,6 +21,8 @@ public class ModuleState
         public List<IAnnotation> Annotations { get; private set; }
         public SyntaxErrorException ModuleException { get; private set; }
         public IDictionary<Tuple<string, DeclarationType>, Attributes> ModuleAttributes { get; private set; }
+        public ConcurrentDictionary<QualifiedModuleName, byte> HasReferenceToModule { get; private set; }
+        public HashSet<QualifiedModuleName> IsReferencedByModule { get; private set; }
```

Comment:
@retailcoder The collection types are not reversed. Since I opted to add the module to module references after all references are resolved, it is done module by module going through all references to the members of the module. That is exactly reverse to the order when resolving the references. So, the `IsReferencedBy` collection is filled from the one thread associated to the module but the members of `HasReferencesTo` get added from different threads.